src/cluster_legacy.c

```diff
@@ -3703,7 +3714,13 @@ void clusterSendPing(clusterLink *link, int type) {
 
     /* Compute the actual total length and send! */
     uint32_t totlen = 0;
-    totlen += writePingExt(hdr, gossipcount);
+
+    if (server.process_clustermsg_extensions) {
+        hdr->mflags[0] |= CLUSTERMSG_FLAG0_NODE_SUPPORTS_EXTENSIONS;
```

Comment:
We should set `CLUSTERMSG_FLAG0_NODE_SUPPORTS_EXTENSIONS` in `writePingExt()`, which is the reason why this flag needs to be sent. Also it is more logical to set this flag only when the receiver supports extension.

Reply:
Thanks for your suggestion. Getting rid of this flag and using `CLUSTERMSG_FLAG0_EXT_DATA`.